src/coprocessor/dag/expr/math.rs

```diff
@@ -0,0 +1,364 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::i64;
+use std::borrow::Cow;
+use coprocessor::codec::Datum;
+use coprocessor::codec::mysql::Decimal;
+use super::{Error, FnCall, Result, StatementContext};
+
+impl FnCall {
+    #[inline]
+    pub fn abs_real(&self, ctx: &StatementContext, row: &[Datum]) -> Result<Option<f64>> {
+        let n = try_opt!(self.children[0].eval_real(ctx, row));
+        Ok(Some(n.abs()))
+    }
+
+    #[inline]
+    pub fn abs_decimal<'a, 'b: 'a>(
+        &'b self,
+        ctx: &StatementContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, Decimal>>> {
+        let d = try_opt!(self.children[0].eval_decimal(ctx, row));
+        let result: Result<Decimal> = d.as_ref().clone().abs().into();
+        result.map(|t| Some(Cow::Owned(t)))
+    }
+
+    #[inline]
+    pub fn abs_int(&self, ctx: &StatementContext, row: &[Datum]) -> Result<Option<i64>> {
+        let n = try_opt!(self.children[0].eval_int(ctx, row));
+        if n == i64::MIN {
+            return Err(Error::Overflow);
+        }
+        Ok(Some(n.abs()))
+    }
+
+    #[inline]
+    pub fn abs_uint(&self, ctx: &StatementContext, row: &[Datum]) -> Result<Option<i64>> {
+        self.children[0].eval_int(ctx, row)
+    }
+
+    #[inline]
+    pub fn ceil_real(&self, ctx: &StatementContext, row: &[Datum]) -> Result<Option<f64>> {
+        let n = try_opt!(self.children[0].eval_real(ctx, row));
+        Ok(Some(n.ceil()))
+    }
+
+    #[inline]
+    pub fn ceil_int_to_int(&self, ctx: &StatementContext, row: &[Datum]) -> Result<Option<i64>> {
+        self.children[0].eval_int(ctx, row)
```

Comment:
When parsing SQL, TiDB won't remove `ceil` operator on `int`, it will calculate that. So it also push `CeilIntIntSig` down to TiKV. Although we know it does nothing, but we should deal with it.

Reply:
Can optimizer optimize this? /cc @hanfei1991